packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -395,6 +400,14 @@ private boolean hasAudioPermission() {
               == PackageManager.PERMISSION_GRANTED;
     }
 
+    private void setBestAERange(CameraCharacteristics characteristics) {
+      Range<Integer>[] fpsRanges = characteristics.get(CameraCharacteristics.CONTROL_AE_AVAILABLE_TARGET_FPS_RANGES);
+      
+      if(fpsRanges.length > 0) {
+        aeFPSRange = fpsRanges[0];
```

Comment:
What makes the first returned range the "best"?

Reply:
This is the slowest FPS off the list, and from what I recall when attempting to fix this issue in my app was that the slowest FPS = better auto exposure result of the preview image. That and the fact that the stock camera app on the device is the same FPS I followed suit.

The range values can come in two different varieties:
- (x,x) Where the FPS ranges are the same (I get 3 of these in that list on my device (7,7), (14,14) and (30,30))
- (x, y) Where there is an allowed leeway for the FPS. But from what I recall when trying to fix this issue in my own project led to sporadic behaviour such as the preview image flicking between dark / normal exposure.

I can take a look again tonight and see if including one of the (x,y) ranges that has the largest range between the x and y component to see if this would produce a better overall frame-rate of the preview but all the while keeping the resulting auto exposure values sane?

Reply:
Hey @NickCullen i have tried your code it solved the problem but when the camera start recording the screen zoom in and become dark again and after the recording is done it zoom outs and become light again. And the file also saved as dark recorded. Please help me out here.

Reply:
Hi @jatinmadaan8 I made a comment a few minutes ago [HERE](https://github.com/flutter/flutter/issues/27767) about using continuous picture mode to use AE and AF settings from the preview. Could you give that a try?

As a side note though I notice you are talking about video here, I have no experience with the video side of things I'm afraid, I can only hope that this will resolve your issue.

Reply:
Hello @NickCullen i tried it today still no progress. I don't know for no reason when the recording starts it gets zoom and dark.